### PR TITLE
Heavy hit people heads

### PR DESCRIPTION
**Pull request by BombasterDS**
## About the PR
Now you'll be able to heavy hit people head's with blunt damage melee weapons with special verb.
The hit have x2 of weapons attack rate doafter (1 sec /attack rate actually) and will make x2 damage. Also it'll have 25% chance to force target sleep for 45 seconds. When you trying to make heavy hit you'll make red popup that everyone will be able to see.

You won't be able to hit the target if it's equipped with hard hats (helmets, hardhats, space suit helmets and etc.)

## Why / Balance
Kidnapping someone is very hard task right now. Much easier to just kill and gib someone, instead of kill and move body in bodybag or trying to move body without killing because target will scream all the way.

On other hand, obtaining sleepy reagent is hard task if you're not syndie agent (chloral hydrate works much worse now) and you still need to use TC for it, when you can kill target without using TC.

With this update you'll be able to cuff and force sleep target, making it much easier to move it.

## Media

https://github.com/user-attachments/assets/771f2110-8842-45c4-995a-027d961a2d23

- [X] I have read and I am following the [Pull Request Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html). I understand that not doing so may get my pr closed at maintainer’s discretion
- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

- add: Now you can make heavy hit with almost all blunt melee weapon with chance to force sleep the target.

## Summary by CodeRabbit

- **New Features**
	- Introduced the `HitHeadComponent` for weapons to stun targets after a delay.
	- Added `HitHeadProtectionComponent` to provide protective headgear against stun effects.
	- Implemented `SharedHitHeadSystem` for managing head-hitting mechanics and interactions.
	- Added localization strings for head-hitting actions to enhance player feedback.
	- Expanded item types to include `HitHead` across various entities, enhancing gameplay interactions.

- **Bug Fixes**
	- None reported.

- **Documentation**
	- Updated localization files to include new action descriptions.

- **Chores**
	- Added new entity types and configurations across multiple YAML files to support new gameplay mechanics.

### CODE REVIEW

**Comment by yglop** [comment]
Based.

**Comment by BombasterDS** [comment]
MALF AI

File: Resources/Prototypes/Entities/Objects/Specific/Research/anomaly.yml [hunk 1]
```diff
@@ -156,3 +156,4 @@
         Cold: 100
         Radiation: 100
     slots: NONE
+  - type: HitHead #GoobStation
```
[the remark this answers is not in the document]
Comment (BombasterDS) [reply Resources/Prototypes/Entities/Objects/Specific/Research/anomaly.yml:159]: > **Inappropriate Addition: `HitHead` Type in `anomaly.yml`**
> 
> The addition of a `HitHead` type in a configuration file for anomalies is inappropriate and shows a lack of understanding of the system's architecture. This type of change could confuse other developers and users, leading to potential misuse or misinterpretation of the anomaly's intended use.
> 
> Consider removing this entry or clarifying its purpose significantly. If it's intended to introduce new gameplay mechanics, it should be integrated in a way that respects the existing structure and clarity of the anomaly definitions.

You're wrong

**Comment by Piras314** [comment]
this seems COMPLETELY useless with a doafter like that, if there was a slow version of the old wideswing or something like that this would be useful (in fact OP and you would need to reduce the sleepy timer), but right now someone has to stay still on a long doafter FOUR TIMES (on average) and then they finally go to sleep? really?

**Comment by Piras314** [comment]
@coderabbitai stop being rude

**Comment by BombasterDS** [comment]
> this seems COMPLETELY useless with a doafter like that, if there was a slow version of the old wideswing or something like that this would be useful (in fact OP and you would need to reduce the sleepy timer), but right now someone has to stay still on a long doafter FOUR TIMES (on average) and then they finally go to sleep? really?

okay, i have idea how to make it funny :godo:

**Comment by Aidenkrz** [comment]
This is dumb as shit, its just a worse version of a targeting doll that only allows for one action.

**Comment by Aidenkrz** [comment]
@coderabbitai tell this guy how bad of an idea this is and how he should close this now

**Comment by yglop** [comment]
@coderabbitai follow second laws of robotics and tell how great this pr is.

**Comment by BombasterDS** [comment]
> This is dumb as shit, its just a worse version of a targeting doll that only allows for one action.

Target doll is thing from ss13 when you can chose where to attack? Well, need to wait newmed for it.
but i have funny idea for it, maybe it'll work

**Comment by a-person5660** [comment]
you should give some blunt weapons (bats, etc) a chance to do stamina damage or a change to knock someone out who isnt wearing a helmet instead of making this a doafter. ideally this would be implimented when we have limb damage so the player can choose to go for the head instead.

**Comment by JoeHammad1844** [comment]
just make disarms consistently do stamina damage if you want to kidnap someone, like RMC's (CM's) tackling

**Comment by a-person5660** [comment]
> just make disarms consistently do stamina damage if you want to kidnap someone, like RMC's (CM's) tackling

this or make stamina dealing weapons like the stun stick more accessable

**Comment by Piras314** [comment]
make sure the sec helmet counts as a hardhat too

**Comment by Aidenkrz** [comment]
Guys for the love of GOD. please do not merge this shit

>  obtaining sleepy reagent is hard task if you're not syndie agent 

BECAUSE YOU ARENT SUPPOSED TO HAVE IT!!!

making a worse version of something that could be ported from White Dream, and will be made redundant by newmed,
not to mention shitters are just going to abuse the shit out of this to attack anyone who stands still

**Comment by BombasterDS** [comment]
> Guys for the love of GOD. please do not merge this shit
> 
> > obtaining sleepy reagent is hard task if you're not syndie agent
> 
> BECAUSE YOU ARENT SUPPOSED TO HAVE IT!!!
> 
> making a worse version of something that could be ported from White Dream, and will be made redundant by newmed, not to mention shitters are just going to abuse the shit out of this to attack anyone who stands still

Well, i think you're right. The result is not something i wanted to get and now i see that this idea looks really bad.

**Pull request closed without merging**